Add information about OpenMP backend to documentation (#421)
oneapi-src/oneDPL: documentation/library_guide/parallel_api/execution_policies.rst

The implementation is based on Parallel STL from the
`LLVM Project <https://github.com/llvm/llvm-project/tree/main/pstl>`_.

|onedpl_short| supports two parallel backends for execution with ``par`` and ``par_unseq`` policies:

> #. *TBB backend* (enabled by default) uses |onetbb_long| or |tbb_long| for parallel execution.

Reviewer: Please remove the italics.

Author: done

> #. *OpenMP backend* uses OpenMP* pragmas for parallel execution. Visit

Reviewer: Please remove the italics.

Author: done

The implementation is based on Parallel STL from the
`LLVM Project <https://github.com/llvm/llvm-project/tree/main/pstl>`_.

> Execution with ``par`` and ``par_unseq`` policies supports two parallel backends:

Reviewer 2: ```suggestion
|onedpl_short| supports two parallel backends for execution with ``par`` and ``par_unseq`` policies:
```

Author: done

> #. `TBB backend` (enabled by default). Means that the algorithms are parallelized with either

Reviewer 2: Why do you use backquotes around the backend names? How will it look when rendered - written with monospace font?

> #. `TBB backend` (enabled by default). Means that the algorithms are parallelized with either

Author: It's rendered as _Italic_. Changed to more appropriate RST syntax for that. If you mind about Italic please let me know

> #. `TBB backend` (enabled by default). When chosen the algorithms are parallelized with either
>    |onetbb_long| or |tbb_long|.

Reviewer 2: ```suggestion
#. `TBB backend` (enabled by default) uses |onetbb_long| or |tbb_long| for parallel execution.
```

Author: done

> #. `OpenMP backend`. When chosen algorithms are parallelized with OpenMP*. Visit
>    :doc:`Macros <../macros>` for the information how to enable OpenMP* backend.

Reviewer 2: ```suggestion
#. `OpenMP backend` uses OpenMP* pragmas for parallel execution. Visit
   :doc:`Macros <../macros>` for the information how to enable the OpenMP backend.
```

Author: done

#. Use the C++ Standard Execution Policies:

   #. Compile the code with options that enable OpenMP* vectorization pragmas.
   #. Either:
      
      * Link with |onetbb_long| or |tbb_long| dynamic library for `TBB backend`

>       * Enable OpenMP* with the dedicated compiler option for `OpenMP backend`

Reviewer 2: I'd change to:
```
* Compile the code with options that enable OpenMP* parallelism and/or vectorization pragmas.
* Link with the |onetbb_long| or |tbb_long| dynamic library for TBB-based parallelism.
```

Author: done

>       * Enable OpenMP* with the dedicated compiler option for `OpenMP backend`

Reviewer 2: I'd change to:
```
* Compile the code with options that enable OpenMP* parallelism and/or vectorization pragmas.
* Link with the |onetbb_long| or |tbb_long| dynamic library for TBB-based parallelism.
```

Author: done
